presto-main/src/main/java/com/facebook/presto/server/security/LdapFilter.java

```diff
@@ -0,0 +1,226 @@
+import javax.naming.AuthenticationException;
+import javax.naming.NamingException;
+import javax.naming.directory.DirContext;
+import javax.naming.directory.InitialDirContext;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletRequestWrapper;
+import javax.servlet.http.HttpServletResponse;
+
+import java.io.IOException;
+import java.security.Principal;
+import java.util.Base64;
+import java.util.Hashtable;
+import java.util.List;
+import java.util.Objects;
+
+import static com.google.common.base.Preconditions.checkState;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+import static javax.naming.Context.INITIAL_CONTEXT_FACTORY;
+import static javax.naming.Context.PROVIDER_URL;
+import static javax.naming.Context.SECURITY_AUTHENTICATION;
+import static javax.naming.Context.SECURITY_CREDENTIALS;
+import static javax.naming.Context.SECURITY_PRINCIPAL;
+
+public class LdapFilter
+        implements Filter
+{
+    private static final Logger LOG = Logger.get(LdapFilter.class);
+    public static final String AUTHENTICATION_TYPE = "basic";
+    public static final String LDAP_CONTEXT_FACTORY = "com.sun.jndi.ldap.LdapCtxFactory";
+    private final String ldapUrl;
+    private final LdapBinder ldapBinder;
+
+    @Inject
+    public LdapFilter(LdapServerConfig config, LdapBinder ldapBinder)
+    {
+        this.ldapUrl = requireNonNull(config.getLdapUrl(), "ldapUrl is null");
+        this.ldapBinder = requireNonNull(ldapBinder, "ldapBinder is null");
+
+        try {
+            authenticate(getBasicEnvironment());
+        }
+        catch (NamingException e) {
+            throw Throwables.propagate(e);
+        }
+    }
+
+    private InitialDirContext authenticate(Hashtable<String, String> environment)
+            throws NamingException
+    {
+        return new InitialDirContext(environment);
+    }
+
+    @Override
+    public void doFilter(ServletRequest servletRequest, ServletResponse servletResponse, FilterChain nextFilter)
+            throws IOException, ServletException
+    {
+        // skip auth for http
+        if (!servletRequest.isSecure()) {
+            LOG.debug("Skipping LDAP authentication in HTTP.");
+            nextFilter.doFilter(servletRequest, servletResponse);
+            return;
+        }
+
+        HttpServletRequest request = (HttpServletRequest) servletRequest;
+        HttpServletResponse response = (HttpServletResponse) servletResponse;
+        String header = request.getHeader(HttpHeaders.AUTHORIZATION);
+
+        if (header != null) {
+            List<String> parts = Splitter.onPattern("\\s+").splitToList(header);
+
+            checkState(parts.size() == 2, "Invalid value for authorization header");
```

Comment:
I think these should return an HTTP 400. How is an exception like `IllegalStateException` thrown from here translated into an HTTP response?